packages/cli/test/unit/openapi/schema-model.unit.js

```diff
@@ -63,6 +69,140 @@ describe('schema to model', () => {
     ]);
   });
 
+  it('generates models for uspto with promoted anonymous schemas', () => {
+    const models = usptoSpecAnonymous.modelSpecs;
+    expect(models).to.eql([
+      {
+        description: 'dataSetList',
+        name: 'dataSetList',
+        className: 'DataSetList',
+        fileName: 'data-set-list.model.ts',
+        properties: [
+          {
+            name: 'total',
+            signature: 'total?: number;',
+            decoration: "@property({name: 'total'})",
+          },
+          {
+            name: 'apis',
+            signature:
+              'apis?: {\n  apiKey?: string;\n  apiVersionNumber?: string;\n  ' +
+              'apiUrl?: string;\n  apiDocumentationUrl?: string;\n}[];',
+            decoration: "@property({name: 'apis'})",
+          },
+        ],
+        imports: [],
+        import: "import {DataSetList} from './data-set-list.model';",
+        kind: 'class',
+        declaration:
+          '{\n  total?: number;\n  apis?: {\n  apiKey?: string;\n  ' +
+          'apiVersionNumber?: string;\n  apiUrl?: string;\n  ' +
+          'apiDocumentationUrl?: string;\n}[];\n}',
+        signature: 'DataSetList',
+      },
+      {
+        description: 'dataSetList',
+        name: 'dataSetList',
+        className: 'DataSetList',
+        fileName: 'data-set-list.model.ts',
+        properties: [
+          {
+            name: 'total',
+            signature: 'total?: number;',
+            decoration: "@property({name: 'total'})",
+          },
+          {
+            name: 'apis',
+            signature:
+              'apis?: {\n  apiKey?: string;\n  apiVersionNumber?: string;\n  ' +
+              'apiUrl?: string;\n  apiDocumentationUrl?: string;\n}[];',
+            decoration: "@property({name: 'apis'})",
+          },
+        ],
+        imports: [],
+        import: "import {DataSetList} from './data-set-list.model';",
+        kind: 'class',
+        declaration:
+          '{\n  total?: number;\n  apis?: {\n  apiKey?: string;\n  ' +
+          'apiVersionNumber?: string;\n  apiUrl?: string;\n  ' +
+          'apiDocumentationUrl?: string;\n}[];\n}',
+        signature: 'DataSetList',
+      },
+      {
+        description: 'AnonymousType_2',
+        name: 'AnonymousType_2',
+        className: 'AnonymousType2',
+        fileName: 'anonymous-type-2.model.ts',
+        properties: [],
+        imports: [],
+        import: "import {AnonymousType2} from './anonymous-type-2.model';",
+        declaration: 'string',
```

Comment:
IIUC, we are defining a custom type (a custom class) to represent a string type? I feel that's an overkill. Also, from my experience, built-in types like `String` and `Number` are difficult to subclass in JavaScript because of edge cases where `String` instance behaves differently from `string` value.

I am proposing to create anonymous types only for objects and arrays, and leave scalar values (strings, numbers, booleans) as anonymous schemas.

Thoughts?

Reply:
+1, that was a mistake.